digital_asset_types/src/dao/mod.rs

```diff
@@ -88,16 +114,36 @@ impl SearchAssetsQuery {
                     .clone()
                     .map(|x| asset::Column::SpecificationVersion.eq(x)),
             )
+            .add_option({
+                self.validation_checks()?;
```

Comment:
We should verify before this point and add_options always or use this check for deciding to include or not. Not for correctness of params.